Added URL provider

Maarten-vd-Sande opened the pull request:
See issue #46 

Two parts:
- some minor cleanups I came across in the code
- I added an URL provider. The URL provider is a barebone Provider class, which just returns the url when `get_genome_download_link` is called. I changed the `get_localname` function to get a localname from an url. I am not sure however if my 'heuristic' approach to spotting an url always works (checking for / . and : ). I could for instance change it so that it checks if `urlopen(name)` works.  

You can now install through:
`genomepy install https://research.nhgri.nih.gov/hydra/download/assembly/Hm105_Dovetail_Assembly_1.0.fa.gz URL`

Is `URL` a good provider name? Capitals or not? Should we be afraid this breaks something somewhere? Should I add tests? The current test already fails at the build stage, does this indicate something is already wrong? Should I update the readme?

simonvh commented:
* Please add a test, yes. Can be basic, just as long as this provider works.
* A solution for the name issue could be that with a URL a localname should always be provided? On the  other hand, this is probably fine as is.
* `URL` is fine. We should probably make the provider name case insensitive anyway. This could easily be done in `ProviderBase.create()`.
* Please update the README, yes, as well as `CHANGELOG.md`.

The test is failing due to some weird issue with conda.

simonvh commented:
Travis config is fixed in the develop branch, c50d920.

Maarten-vd-Sande commented:
@simonvh 
- I added some simple tests
- I changed that urllib checks if an url is a valid url
- Providers are case insensitive now
- README and changelog are updated
